agent/stats/task_windows.go

```diff
@@ -17,64 +18,42 @@ package stats
 
 import (
 	"context"
-	"os/exec"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/eni/networkutils"
 	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
 
 	dockerstats "github.com/docker/docker/api/types"
 	"github.com/pkg/errors"
 )
 
-const (
-	receivedBroadcastPackets = "ReceivedBroadcastPackets"
-	receivedMulticastPackets = "ReceivedMulticastPackets"
-	receivedUnicastPackets   = "ReceivedUnicastPackets"
-	sentBroadcastPackets     = "SentBroadcastPackets"
-	sentMulticastPackets     = "SentMulticastPackets"
-	sentUnicastPackets       = "SentUnicastPackets"
-	receivedBytes            = "ReceivedBytes"
-	receivedPacketErrors     = "ReceivedPacketErrors"
-	receivedDiscardedPackets = "ReceivedDiscardedPackets"
-	sentBytes                = "SentBytes"
-	outboundPacketErrors     = "OutboundPacketErrors"
-	outboundDiscardedPackets = "OutboundDiscardedPackets"
-)
-
-var (
-	// Making it visible for unit testing
-	execCommand = exec.Command
-	// Fields to be extracted from the stats returned by cmdlet.
-	networkStatKeys = []string{
-		receivedBroadcastPackets,
-		receivedMulticastPackets,
-		receivedUnicastPackets,
-		sentBroadcastPackets,
-		sentMulticastPackets,
-		sentUnicastPackets,
-		receivedBytes,
-		receivedPacketErrors,
-		receivedDiscardedPackets,
-		sentBytes,
-		outboundDiscardedPackets,
-		outboundPacketErrors,
-	}
-)
-
 type StatsTask struct {
 	*statsTaskCommon
+	interfaceLUID []uint64
+	netUtils      networkutils.NetworkUtils
 }
 
 func newStatsTaskContainer(taskARN, taskId, containerPID string, numberOfContainers int,
 	resolver resolver.ContainerMetadataResolver, publishInterval time.Duration, taskENIs task.TaskENIs) (*StatsTask, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 
+	// Instantiate an instance of network utils.
+	// This interface would be used to invoke Windows networking APIs.
+	netUtils := networkutils.New()
+
 	devices := make([]string, len(taskENIs))
+	ifaceLUID := make([]uint64, len(taskENIs))
```

Comment:
You can have multiple eni's per task?

Reply:
That is the case with trunk ENI I think.